CanHazFunny/CanHazFunny/Jester.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+
+namespace CanHazFunny
+{
+    public class Jester
+    {
+        private IJokeService JokeService;
+        private IJokeOutput JokeOutServ;
+
+        public IJokeOutput JokeOut { get => JokeOutServ; set => JokeOutServ = value; }
+        public IJokeService JokeServ { get => JokeService; set => JokeService = value; }
+
+        public Jester(IJokeService jokeService, IJokeOutput outJoke)
+        {
+            if(jokeService == null || outJoke == null)
+            {
+                throw new ArgumentNullException(nameof(jokeService));
+            }
+            else
+            {
+                this.JokeOut = outJoke;
+                this.JokeService = jokeService;
```

Comment:
This is actually assigning the field not the property. This is why we favor just using auto properties rather than always having backing fields for them.
```suggestion
                this.JokeServ = jokeService;
```